maskrcnn_benchmark/utils/metric_logger.py

```diff
@@ -64,3 +68,41 @@ def __str__(self):
                 "{}: {:.4f} ({:.4f})".format(name, meter.median, meter.global_avg)
             )
         return self.delimiter.join(loss_str)
+
+
+class TensorboardLogger(MetricLogger):
+    def __init__(self,
+                 log_dir,
+                 start_iter=0,
+                 delimiter='\t'):
+
+        super(TensorboardLogger, self).__init__(delimiter)
+        self.iteration = start_iter
+        self.writer = self._get_tensorboard_writer(log_dir)
+
+    @staticmethod
+    def _get_tensorboard_writer(log_dir):
+        try:
+            from tensorboardX import SummaryWriter
+        except ImportError:
+            raise ImportError(
+                'To use tensorboard please install tensorboardX '
+                '[ pip install tensorflow tensorboardX ].'
+            )
+
+        if is_main_process():
+            timestamp = datetime.fromtimestamp(time.time()).strftime('%Y%m%d-%H:%M')
+            tb_logger = SummaryWriter('{}-{}'.format(log_dir, timestamp))
+            return tb_logger
+        else:
+            return None
+
+    def update(self, ** kwargs):
+        super(TensorboardLogger, self).update(**kwargs)
+        if self.writer:
+            for k, v in kwargs.items():
+                if isinstance(v, torch.Tensor):
+                    v = v.item()
+                assert isinstance(v, (float, int))
+                self.writer.add_scalar(k, v, self.iteration)
+            self.iteration += 1
```

Comment:
self.iteration += 1
will cause the iterations logged in tensorboard inconsistent with the training iterations. As in the trainer.py, there are two meters.update within one iteration. From my opinion, it may be better to get self.iteration from the arguments iteration or arguments["iteration"] in do_train() in trainer.py.

Reply:
@miguelvr